### DISCUSSION TITLE
What is the difference between outline and structure?

### DISCUSSION THREAD

**Discussion by septatrix** [James-Yu/LaTeX-Workshop#3238] [Q&A] [answered]
VSCode provides the "outline" panel under the "Explorer" view whereas LaTeX-Workshop also provides the "structure" panel under the "LaTeX" view.

At a first look they seem almost identical.
The only difference I was able to find is that the outline panel shows little text icons next to the entries and supports filtering as well as sorting.

To me it seems like the custom structure panel has little benefit over the native outline panel or have I maybe overlooked some secret feature?

**Comment by jlelong** [comment, marked as answer]
The _outline_ view in the _Explorer_ panel provides access to the structure of the active document only. Any included file is ignored.

The _structure_ view in the _TeX_ panel provides access to the structure of the whole LaTeX project starting from the root file and going through all the included files recursively.

**Reply by septatrix** [reply]
Oh that is neat. So I guess the outline view just does syntactical analysis on the document whereas the structure view presumably uses some of the generated intermediate files for this information? Or is it simply hardcoded to follow includes and inputs? Nevertheless thanks for clearing that up. I guess I work to seldomly with documents split over multiple  files :D

**Reply by jlelong** [reply]
No, it does not use any intermediate generated files. The structure view just follows `\include`, `\input` and any related commands.
